### PR TITLE
EVG-2663 fixup rest models for config

### PR DESCRIPTION
**Pull request by john-m-liu**
I made everything nullable because I want the requester to be able to omit fields they don't want changed

### CODE REVIEW

File: rest/model/admin.go [hunk 1]
```diff
@@ -6,38 +6,63 @@ import (
...
-	LoggerConfig       APILoggerConfig                   `json:"logger_config"`
-	LogPath            APIString                         `json:"log_path"`
-	NewRelic           APINewRelicConfig                 `json:"new_relic"`
-	Notify             APINotifyConfig                   `json:"notify"`
+	Alerts             *APIAlertsConfig                  `json:"alerts"`
+	Amboy              *APIAmboyConfig                   `json:"amboy"`
+	Api                *APIapiConfig                     `json:"api"`
+	ApiUrl             *string                           `json:"api_url"`
+	AuthConfig         *APIAuthConfig                    `json:"auth"`
+	Banner             *string                           `json:"banner"`
```
Comment (richardsamuels) [inline rest/model/admin.go]: Don't these still need to be APIString?

Comment (john-m-liu) [reply rest/model/admin.go]: I don't believe so - the only difference between APIstring and string is that "null" will unmarshal to a pointer to a nil value, which I don't need with this change

**Comment by tychoish** [comment]
should we: 

- convert all APIString to *string just for consistency
- implement this change by adding some functionality to APIString
- Could we implement fmt.Stringer for APIString and have APIString be an alias for *string

**Comment by john-m-liu** [comment]
I took a shot at changing APIString, but it was incredibly messy and where I'm at it still is far from compiling. I'd like to keep it like this because *string seems like a valid alternative to APIString

**Comment by tychoish** [comment]
lgtm iff you open a ticket to cleanup APIString in the near future and put it in the 2018-03-12 sprint

**Comment by john-m-liu** [comment]
EVG-2770

**Pull request merged**